lib/codegen/Decision.cpp

```diff
@@ -520,12 +523,12 @@ void abortWhenStuck(llvm::BasicBlock *CurrentBlock, llvm::Module *Module, KORESy
   symbol = d->getAllSymbols().at(Out.str());
   auto BlockType = getBlockType(Module, d, symbol);
   llvm::Value *Ptr;
-  auto BlockPtr = llvm::PointerType::getUnqual(getTypeByName(Module, BLOCK_STRUCT));
+  auto BlockPtr = llvm::PointerType::get(getTypeByName(Module, BLOCK_STRUCT), 1);
   if (symbol->getArguments().empty()) {
-    Ptr = llvm::ConstantExpr::getIntToPtr(llvm::ConstantInt::get(llvm::Type::getInt64Ty(Ctx), ((uint64_t)symbol->getTag() << 32 | 1)), getValueType({SortCategory::Symbol, 0}, Module));
+    Ptr = llvm::CallInst::Create(getOrInsertFunction(Module, "inttoptr_i64.p1s_blocks", getValueType({SortCategory::Symbol, 0}, Module), llvm::Type::getInt64Ty(Ctx)), {llvm::ConstantInt::get(llvm::Type::getInt64Ty(Ctx), (((uint64_t)symbol->getTag()) << 32) | 1)}, "", CurrentBlock);
```

Comment:
Why doesn't this fit the int-to-pointer conversion functions defined previously?

Reply:
I'm not sure what you mean.

Reply:
Ah sorry - I saw `llvm::Value *ptrToInt(llvm::Value *val, llvm::BasicBlock *block);` defined above and mistakenly assumed that you had also implemented `intToPtr` as a similar helper.

Reply:
It was easier to just inline this one because it is the only occurrence in the code base and always has the same types.